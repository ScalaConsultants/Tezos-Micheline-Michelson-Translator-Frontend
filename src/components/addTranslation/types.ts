export type AddTranslationState = {
  setShowModal: Function;
};

export type FormValues = {
  title: string;
  author: string;
  description: string;
  isEmail: string;
  email: string;
};
