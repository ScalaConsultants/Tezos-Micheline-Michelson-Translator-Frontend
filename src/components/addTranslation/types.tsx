export type AddTranslationState = {
  setShowModal: Function;
};

export type FormValues = {
  name: string;
  author: string;
  description: string;
};
