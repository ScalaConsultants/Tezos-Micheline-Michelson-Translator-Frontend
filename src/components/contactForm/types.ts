export type FormValues = {
  name: string;
  phone?: string;
  email?: string;
  content: string;
};

export type ValidationErrors = {
  name?: string;
  phone?: string;
  email?: string;
  content?: string;
};

export type FormikSubmitting = {
  setSubmitting: Function;
};
