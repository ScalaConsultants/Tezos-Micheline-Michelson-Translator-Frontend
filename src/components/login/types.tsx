import { authState } from "../../store/authentication/types";

export interface IState {
  pending: boolean;
  auth: authState;
  error?: string;
}

export type FormValues = {
  login: string;
  password: string;
};
