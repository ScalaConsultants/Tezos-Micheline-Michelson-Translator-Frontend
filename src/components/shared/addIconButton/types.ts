export type AddIconButtonProps = {
  type?: "submit" | "reset" | "button";
  label?: string;
  onClick?: any;
  disabled?: boolean;
  img?: string;
};
