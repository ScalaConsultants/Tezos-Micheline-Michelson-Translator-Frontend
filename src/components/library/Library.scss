@import "../../scss/vars";

.Library {
  &__header {
    color: $text-white;
    background: $background-color-grey-3;
    border-radius: 3px 3px 0 0;
    padding: 10px;
    height: 30px;
    font-family: Roboto;
    font-style: normal;
    font-weight: 500;
    font-size: 12px;
    line-height: 14px;
    display: flex;
    flex-wrap: wrap;
    align-items: center;
    text-align: center;
    justify-content: center;
    letter-spacing: 0.15px;
    text-transform: uppercase;

    button {
      background: transparent;
      border: 0;
      border-bottom: 2px solid $border-grey3;
      padding: 11px;
      font-family: Roboto;
      font-style: normal;
      font-weight: 500;
      font-size: 12px;
      line-height: 14px;
      display: flex;
      align-items: center;
      text-align: center;
      letter-spacing: 0.15px;
      text-transform: uppercase;
      color: #FFFFFF;
      outline: 0;

      &:hover {
        cursor: pointer;
        border-bottom: 2px solid $border-white;
      }
    }

    img {
      display: flex;
    }

    &-selected {
        border-bottom: 2px solid $border-red !important;
    }

    &__translate-button {
      margin-left: 40px;
      margin-right: 40px;
      cursor: pointer;
    }

    @media screen and (max-width: 500px) {
      height: 150px;

      button, img {
        flex: 0 0 100%;
        flex-wrap: wrap;
        display: block;
      }
    }
  }
}
