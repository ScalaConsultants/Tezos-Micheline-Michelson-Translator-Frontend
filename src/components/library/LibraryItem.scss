@import "../../scss/vars";

.LibraryItem {
    box-shadow: 0px 1px 3px rgba(0, 0, 0, 0.2), 0px 2px 2px rgba(0, 0, 0, 0.12), 0px 0px 2px rgba(0, 0, 0, 0.14);
    border-radius: 3px;
    height: auto;
    margin-bottom: 30px;
    align-items: center;
    vertical-align: top;
    resize: none;
    box-sizing: border-box;
    color: $text-grey1;
    font-family: Roboto;
    font-style: normal;
    font-weight: normal;
    font-size: 12px;
    line-height: 17px;
    letter-spacing: 0.15px;

    @media screen and (max-width: 500px) {
        flex: 0 0 100%;
        &:nth-child(even) {
            background-color: $background-color-grey-light;
        }
    }

    &__info-area-wrapper {
        box-sizing: border-box;
        width: 50%;
        height: 140px;
        display: inline-flex;
        flex-direction: row;
        flex-wrap: wrap;
        font-family: Roboto;
        font-style: normal;
        font-weight: 300;
        font-size: 20px;
        line-height: 23px;
        letter-spacing: 0.15px;
        color: $text-dark2;
        padding: 10px;
        vertical-align: top;

        @media screen and (max-width: 500px) {
            width: 100%;
            height: auto;
            display: flex;
        }
    }

    &__detail-wrapper {
        box-sizing: border-box;
        width: 50%;
        height: 60px;
        display: inline-flex;
        flex-direction: column;
        font-family: Roboto;
        font-style: normal;
        font-weight: 300;
        font-size: 20px;
        line-height: 23px;
        letter-spacing: 0.15px;
        color: $text-dark2;

        &--label {
            display: flex;
            font-family: Roboto;
            font-style: normal;
            font-weight: 300;
            font-size: 20px;
            line-height: 23px;
            letter-spacing: 0.15px;
            color: $text-dark2;
            padding: 5px;
        }

        &--content {
            display: flex;
            font-family: Roboto;
            font-style: normal;
            font-weight: 500;
            font-size: 14px;
            line-height: 16px;
            letter-spacing: 0.15px;
            color: $text-dark2;
            opacity: 0.5;
            padding: 5px;
        }

        @media screen and (max-width: 500px) {
            width: 100%;
            display: flex;
        }
    }

    &__description-wrapper {
        box-sizing: border-box;
        width: 100%;
        height: 120px;
        display: inline-flex;
        flex-direction: column;
        font-family: Roboto;
        font-style: normal;
        font-weight: 300;
        font-size: 20px;
        line-height: 23px;
        letter-spacing: 0.15px;
        color: $text-dark2;

        &--label {
            display: flex;
            font-family: Roboto;
            font-style: normal;
            font-weight: 300;
            font-size: 20px;
            line-height: 23px;
            letter-spacing: 0.15px;
            color: $text-dark2;
            padding: 5px;
        }

        &--content {
            display: flex;
            font-family: Roboto;
            font-style: normal;
            font-weight: 500;
            font-size: 12px;
            line-height: 14px;
            letter-spacing: 0.15px;

            color: $text-dark2;

            opacity: 0.5;
            padding: 5px;
        }

        @media screen and (max-width: 500px) {
            width: 100%;
            display: flex;
        }
    }


    &__wrapper2 {
        box-sizing: border-box;
        width: 50%;
        height: 200px;
        display: inline-flex;
        flex-direction: column;
        font-family: Roboto;
        font-style: normal;
        font-weight: 300;
        font-size: 20px;
        line-height: 23px;
        letter-spacing: 0.15px;
        color: $text-dark2;
        padding: 10px;
        cursor: pointer;

        &:hover {
            &:nth-of-type(3) {
                background-color: $background-color-grey-2;
            }

            &:nth-of-type(4) {
                background-color: $background-color-grey-light1;
            }
        }

        &:nth-of-type(3) {
            background-color: $background-color-grey-light2;
        }

        &:nth-of-type(4) {
            background-color: $background-color-grey-light3;
        }

        @media screen and (max-width: 500px) {
            width: 100%;
            display: flex;
        }
    }

    &__label {
        display: flex;
        font-family: Roboto;
        font-style: normal;
        font-weight: 500;
        font-size: 12px;
        line-height: 14px;
        display: flex;
        align-items: center;
        letter-spacing: 0.15px;
        text-transform: uppercase;

        color: $text-grey1;
        padding: 5px;
        border-bottom: 1px solid $border-grey-light4;
    }

    &__content {
        display: flex;
        width: 100%;
        color: $text-grey1;
        font-family: Roboto;
        font-style: normal;
        font-weight: normal;
        font-size: 12px;
        line-height: 17px;
        letter-spacing: 0.15px;
        padding: 10px;
    }
}
