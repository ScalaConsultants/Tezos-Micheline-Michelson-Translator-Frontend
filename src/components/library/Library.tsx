import React from "react";
import { useDispatch, useMappedState } from "redux-react-hook";
import "./Library.scss";
import LibraryItem from "./LibraryItem";
import { translate } from "../translator/Translator";
import * as TranslatorTypes from "../../store/translator/types";
import { IState } from "../../store/global/types";
import * as TranslatorActions from "../../store/translator/actions";
import {bindActionCreators} from "redux";

const mapState = (state: IState) => ({
  library: state.library,
});

const Library = () => {
  const dispatch = useDispatch();
  const { library } = useMappedState(mapState);
  const boundTranslatorActions = bindActionCreators(TranslatorActions, dispatch);

  const selectMicheline = (value: string) => {
    boundTranslatorActions.TranslatorSetMode(TranslatorTypes.Modes.MICHELINEMICHELSON);
    boundTranslatorActions.TranslatorSetMicheline(null, value);
    translate(TranslatorTypes.Modes.MICHELINEMICHELSON, value, boundTranslatorActions);
  };

  const selectMichelson = (value: string) => {
    boundTranslatorActions.TranslatorSetMode(TranslatorTypes.Modes.MICHELSONMICHELINE);
    boundTranslatorActions.TranslatorSetMichelson(null, value);
    translate(TranslatorTypes.Modes.MICHELSONMICHELINE, value, boundTranslatorActions);
  };

  return (
    <div className="Library">
      {library.map((item, key) => (
        <LibraryItem
          selectMicheline={selectMicheline}
          selectMichelson={selectMichelson}
          data={item}
          key={`${item.uid}${key.toString()}`}
        />
      ))}
    </div>
  );
};

export default Library;
