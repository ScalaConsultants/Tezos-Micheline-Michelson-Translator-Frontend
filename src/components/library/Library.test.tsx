import React from "react";
import { mount } from "enzyme/build";
import { StoreContext } from "redux-react-hook";
import Library from "./Library";

import configureStore from "../../store/index";
import rootSaga from "../../store/saga";
import * as LibraryTypes from "../../store/library/types";
import LibraryItem from "./LibraryItem";

const store = configureStore();
store.runSaga(rootSaga);

const library = [
  {
    source:
      "parameter int;\nstorage int;\ncode { CAR ;\n       PUSH int 1 ;\n       ADD ;\n       NIL operation ;\n       PAIR }",
    translation:
      '[ { "prim": "parameter", "args": [ { "prim": "int" } ] }, { "prim": "storage", "args": [ { "prim": "int" } ] }, { "prim": "code", "args": [ [ { "prim": "CAR" },{ "prim": "PUSH", "args": [ { "prim": "int" }, { "int": "1" } ] },{ "prim": "ADD" },{ "prim": "NIL", "args": [ { "prim": "operation" } ] },{ "prim": "PAIR" } ] ] } ]\n',
  },
  {
    source:
      "parameter int;\nstorage int;\ncode { CAR ;\n       PUSH int 1 ;\n       ADD ;\n       NIL operation ;\n       PAIR }",
    translation:
      '[ { "prim": "parameter", "args": [ { "prim": "int" } ] }, { "prim": "storage", "args": [ { "prim": "int" } ] }, { "prim": "code", "args": [ [ { "prim": "CAR" },{ "prim": "PUSH", "args": [ { "prim": "int" }, { "int": "1" } ] },{ "prim": "ADD" },{ "prim": "NIL", "args": [ { "prim": "operation" } ] },{ "prim": "PAIR" } ] ] } ]\n',
  },
  {
    source:
      '[\n    {\n      "prim": "parameter",\n      "args": [\n        {\n          "prim": "int"\n        }\n      ]\n    },\n    {\n      "prim": "storage",\n      "args": [\n        {\n          "prim": "int"\n        }\n      ]\n    },\n    {\n      "prim": "code",\n      "args": [\n        [\n          {\n            "prim": "CAR"\n          },\n          {\n            "prim": "PUSH",\n            "args": [\n              {\n                "prim": "int"\n              },\n              {\n                "int": "1"\n              }\n            ]\n          },\n          {\n            "prim": "ADD"\n          },\n          {\n            "prim": "NIL",\n            "args": [\n              {\n                "prim": "operation"\n              }\n            ]\n          },\n          {\n            "prim": "PAIR"\n          }\n        ]\n      ]\n    }\n]',
    translation:
      "parameter int;\nstorage int;\ncode { CAR ;\n       PUSH int 1 ;\n       ADD ;\n       NIL operation ;\n       PAIR }",
  },
  { source: "EQ", translation: '{ "prim": "EQ" }\n' },
  {
    source:
      '[\n    {\n      "prim": "parameter",\n      "args": [\n        {\n          "prim": "int"\n        }\n      ]\n    },\n    {\n      "prim": "storage",\n      "args": [\n        {\n          "prim": "int"\n        }\n      ]\n    },\n    {\n      "prim": "code",\n      "args": [\n        [\n          {\n            "prim": "CAR"\n          },\n          {\n            "prim": "PUSH",\n            "args": [\n              {\n                "prim": "int"\n              },\n              {\n                "int": "1"\n              }\n            ]\n          },\n          {\n            "prim": "ADD"\n          },\n          {\n            "prim": "NIL",\n            "args": [\n              {\n                "prim": "operation"\n              }\n            ]\n          },\n          {\n            "prim": "PAIR"\n          }\n        ]\n      ]\n    }\n]',
    translation:
      "parameter int;\nstorage int;\ncode { CAR ;\n       PUSH int 1 ;\n       ADD ;\n       NIL operation ;\n       PAIR }",
  },
  {
    source:
      '[\n    {\n      "prim": "parameter",\n      "args": [\n        {\n          "prim": "int"\n        }\n      ]\n    },\n    {\n      "prim": "storage",\n      "args": [\n        {\n          "prim": "int"\n        }\n      ]\n    },\n    {\n      "prim": "code",\n      "args": [\n        [\n          {\n            "prim": "CAR"\n          },\n          {\n            "prim": "PUSH",\n            "args": [\n              {\n                "prim": "int"\n              },\n              {\n                "int": "1"\n              }\n            ]\n          },\n          {\n            "prim": "ADD"\n          },\n          {\n            "prim": "NIL",\n            "args": [\n              {\n                "prim": "operation"\n              }\n            ]\n          },\n          {\n            "prim": "PAIR"\n          }\n        ]\n      ]\n    }\n]',
    translation:
      "parameter int;\nstorage int;\ncode { CAR ;\n       PUSH int 1 ;\n       ADD ;\n       NIL operation ;\n       PAIR }",
  },
  {
    source:
      '[\n  {\n    "prim": "parameter",\n    "args": [\n      {\n        "prim": "int"\n      }\n    ]\n  },\n  {\n    "prim": "storage",\n    "args": [\n      {\n        "prim": "int"\n      }\n    ]\n  },\n  {\n    "prim": "code",\n    "args": [\n      [\n        {\n          "prim": "CAR"\n        },\n        {\n          "prim": "PUSH",\n          "args": [\n            {\n              "prim": "int"\n            },\n            {\n              "int": "1"\n            }\n          ]\n        },\n        {\n          "prim": "ADD"\n        },\n        {\n          "prim": "NIL",\n          "args": [\n            {\n              "prim": "operation"\n            }\n          ]\n        },\n        {\n          "prim": "PAIR"\n        }\n      ]\n    ]\n  }\n]',
    translation:
      "parameter int;\nstorage int;\ncode { CAR ;\n       PUSH int 1 ;\n       ADD ;\n       NIL operation ;\n       PAIR }",
  },
];

it("App -> renders without crashing", () => {
  store.dispatch({
    type: LibraryTypes.LIBRARY_SET,
    payload: library,
  });

  const wrapper = mount(
    <StoreContext.Provider value={store}>
      <Library />
    </StoreContext.Provider>,
  );

  const item = wrapper.find(LibraryItem);
  // expect(item.at(0).find(".LibraryItem__detail-wrapper--label").at(0));
  expect(
    item
      .at(0)
      .find(".LibraryItem__detail-wrapper--label")
      .at(0)
      .text(),
  ).toEqual("Name");

  expect(
    item
      .at(0)
      .find(".LibraryItem__detail-wrapper--content")
      .at(0)
      .text(),
  ).toEqual("some name");

  expect(item.at(0).props().data).toEqual(library[1]);
  expect(item).toHaveLength(7);
});
