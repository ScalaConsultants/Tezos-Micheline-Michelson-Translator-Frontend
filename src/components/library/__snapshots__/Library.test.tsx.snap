// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Library matches snapshot 1`] = `
"<Library>
  <div className=\\"Library\\">
    <LibraryItem selectMicheline={[Function: selectMicheline]} selectMichelson={[Function: selectMichelson]} data={{...}}>
      <div className=\\"LibraryItem\\">
        <div className=\\"LibraryItem__info-area-wrapper\\">
          <div className=\\"LibraryItem__detail-wrapper\\">
            <div className=\\"LibraryItem__detail-wrapper--label\\">
              Title
            </div>
            <div className=\\"LibraryItem__detail-wrapper--content\\">
              Test1
            </div>
          </div>
        </div>
        <div className=\\"LibraryItem__info-area-wrapper\\">
          <div className=\\"LibraryItem__description-wrapper\\">
            <div className=\\"LibraryItem__description-wrapper--label\\">
              Description
            </div>
            <div className=\\"LibraryItem__description-wrapper--content\\">
              Description1
            </div>
          </div>
        </div>
        <div className=\\"LibraryItem__wrapper2\\" onClick={[Function: onClick]}>
          <div className=\\"LibraryItem__label\\">
            Micheline
          </div>
          <div className=\\"LibraryItem__content\\">
            [
              {
                &quot;prim&quot;: &quot;parameter&quot;,
                &quot;args&quot;: [
                  {
                    &quot;prim&quot;: &quot;int&quot;
                  }
                ]
              },
              {
                &quot;prim&quot;: &quot;storage&quot;,
                &quot;args&quot;: [
                  {
                    &quot;prim&quot;: &quot;int&quot;
                  }
                ]
              },
              {
                &quot;prim&quot;: &quot;code&quot;,
                &quot;args&quot;: [
                  [
                    {
                      &quot;prim&quot;: &quot;CAR&quot;
                    },
                    {
                      &quot;prim&quot;: &quot;PUSH&quot;,
                      &quot;args&quot;: [
                        {
                          &quot;prim&quot;: &quot;int&quot;
                        },
                        {
                          &quot;int&quot;: &quot;1&quot;
                        }
                      ]
                    },
                    {
                      &quot;prim&quot;: &quot;ADD&quot;
                    },
                    {
                      &quot;prim&quot;: &quot;NIL&quot;,
                      &quot;args&quot;: [
                        {
                          &quot;prim&quot;: &quot;operation&quot;
                        }
                      ]
                    },
                    {
                      &quot;prim&quot;: &quot;PAIR&quot;
                    }
                  ]
                ]
              }
            ]
          </div>
        </div>
        <div className=\\"LibraryItem__wrapper2\\" onClick={[Function: onClick]}>
          <div className=\\"LibraryItem__label\\">
            Michelson
          </div>
          <div className=\\"LibraryItem__content\\">
            parameter int;
            storage int;
            code { CAR ;
                   PUSH int 1 ;
                   ADD ;
                   NIL operation ;
                   PAIR }
          </div>
        </div>
      </div>
    </LibraryItem>
    <LibraryItem selectMicheline={[Function: selectMicheline]} selectMichelson={[Function: selectMichelson]} data={{...}}>
      <div className=\\"LibraryItem\\">
        <div className=\\"LibraryItem__info-area-wrapper\\">
          <div className=\\"LibraryItem__detail-wrapper\\">
            <div className=\\"LibraryItem__detail-wrapper--label\\">
              Title
            </div>
            <div className=\\"LibraryItem__detail-wrapper--content\\">
              dsddsdsds
            </div>
          </div>
        </div>
        <div className=\\"LibraryItem__info-area-wrapper\\">
          <div className=\\"LibraryItem__description-wrapper\\">
            <div className=\\"LibraryItem__description-wrapper--label\\">
              Description
            </div>
            <div className=\\"LibraryItem__description-wrapper--content\\">
              dsdsdsdsdsds
            </div>
          </div>
        </div>
        <div className=\\"LibraryItem__wrapper2\\" onClick={[Function: onClick]}>
          <div className=\\"LibraryItem__label\\">
            Micheline
          </div>
          <div className=\\"LibraryItem__content\\">
            [
              {
                &quot;prim&quot;: &quot;parameter&quot;,
                &quot;args&quot;: [
                  {
                    &quot;prim&quot;: &quot;int&quot;
                  }
                ]
              },
              {
                &quot;prim&quot;: &quot;storage&quot;,
                &quot;args&quot;: [
                  {
                    &quot;prim&quot;: &quot;int&quot;
                  }
                ]
              },
              {
                &quot;prim&quot;: &quot;code&quot;,
                &quot;args&quot;: [
                  [
                    {
                      &quot;prim&quot;: &quot;CAR&quot;
                    },
                    {
                      &quot;prim&quot;: &quot;PUSH&quot;,
                      &quot;args&quot;: [
                        {
                          &quot;prim&quot;: &quot;int&quot;
                        },
                        {
                          &quot;int&quot;: &quot;1&quot;
                        }
                      ]
                    },
                    {
                      &quot;prim&quot;: &quot;ADD&quot;
                    },
                    {
                      &quot;prim&quot;: &quot;NIL&quot;,
                      &quot;args&quot;: [
                        {
                          &quot;prim&quot;: &quot;operation&quot;
                        }
                      ]
                    },
                    {
                      &quot;prim&quot;: &quot;PAIR&quot;
                    }
                  ]
                ]
              }
            ]
          </div>
        </div>
        <div className=\\"LibraryItem__wrapper2\\" onClick={[Function: onClick]}>
          <div className=\\"LibraryItem__label\\">
            Michelson
          </div>
          <div className=\\"LibraryItem__content\\">
            parameter int;
            storage int;
            code { CAR ;
                   PUSH int 1 ;
                   ADD ;
                   NIL operation ;
                   PAIR }
          </div>
        </div>
      </div>
    </LibraryItem>
    <LibraryItem selectMicheline={[Function: selectMicheline]} selectMichelson={[Function: selectMichelson]} data={{...}}>
      <div className=\\"LibraryItem\\">
        <div className=\\"LibraryItem__info-area-wrapper\\">
          <div className=\\"LibraryItem__detail-wrapper\\">
            <div className=\\"LibraryItem__detail-wrapper--label\\">
              Title
            </div>
            <div className=\\"LibraryItem__detail-wrapper--content\\">
              test
            </div>
          </div>
        </div>
        <div className=\\"LibraryItem__info-area-wrapper\\">
          <div className=\\"LibraryItem__description-wrapper\\">
            <div className=\\"LibraryItem__description-wrapper--label\\">
              Description
            </div>
            <div className=\\"LibraryItem__description-wrapper--content\\">
              sdkofmwofmnwof
            </div>
          </div>
        </div>
        <div className=\\"LibraryItem__wrapper2\\" onClick={[Function: onClick]}>
          <div className=\\"LibraryItem__label\\">
            Micheline
          </div>
          <div className=\\"LibraryItem__content\\">
            [
              {
                &quot;prim&quot;: &quot;parameter&quot;,
                &quot;args&quot;: [
                  {
                    &quot;prim&quot;: &quot;int&quot;
                  }
                ]
              },
              {
                &quot;prim&quot;: &quot;storage&quot;,
                &quot;args&quot;: [
                  {
                    &quot;prim&quot;: &quot;int&quot;
                  }
                ]
              },
              {
                &quot;prim&quot;: &quot;code&quot;,
                &quot;args&quot;: [
                  [
                    {
                      &quot;prim&quot;: &quot;CAR&quot;
                    },
                    {
                      &quot;prim&quot;: &quot;PUSH&quot;,
                      &quot;args&quot;: [
                        {
                          &quot;prim&quot;: &quot;int&quot;
                        },
                        {
                          &quot;int&quot;: &quot;1&quot;
                        }
                      ]
                    },
                    {
                      &quot;prim&quot;: &quot;ADD&quot;
                    },
                    {
                      &quot;prim&quot;: &quot;NIL&quot;,
                      &quot;args&quot;: [
                        {
                          &quot;prim&quot;: &quot;operation&quot;
                        }
                      ]
                    },
                    {
                      &quot;prim&quot;: &quot;PAIR&quot;
                    }
                  ]
                ]
              }
            ]
          </div>
        </div>
        <div className=\\"LibraryItem__wrapper2\\" onClick={[Function: onClick]}>
          <div className=\\"LibraryItem__label\\">
            Michelson
          </div>
          <div className=\\"LibraryItem__content\\">
            parameter int;
            storage int;
            code { CAR ;
                   PUSH int 1 ;
                   ADD ;
                   NIL operation ;
                   PAIR }
          </div>
        </div>
      </div>
    </LibraryItem>
    <LibraryItem selectMicheline={[Function: selectMicheline]} selectMichelson={[Function: selectMichelson]} data={{...}}>
      <div className=\\"LibraryItem\\">
        <div className=\\"LibraryItem__info-area-wrapper\\">
          <div className=\\"LibraryItem__detail-wrapper\\">
            <div className=\\"LibraryItem__detail-wrapper--label\\">
              Title
            </div>
            <div className=\\"LibraryItem__detail-wrapper--content\\">
              dqedfeq
            </div>
          </div>
        </div>
        <div className=\\"LibraryItem__info-area-wrapper\\">
          <div className=\\"LibraryItem__description-wrapper\\">
            <div className=\\"LibraryItem__description-wrapper--label\\">
              Description
            </div>
            <div className=\\"LibraryItem__description-wrapper--content\\">
              qwdq
            </div>
          </div>
        </div>
        <div className=\\"LibraryItem__wrapper2\\" onClick={[Function: onClick]}>
          <div className=\\"LibraryItem__label\\">
            Micheline
          </div>
          <div className=\\"LibraryItem__content\\">
            [
              {
                &quot;prim&quot;: &quot;parameter&quot;,
                &quot;args&quot;: [
                  {
                    &quot;prim&quot;: &quot;int&quot;
                  }
                ]
              },
              {
                &quot;prim&quot;: &quot;storage&quot;,
                &quot;args&quot;: [
                  {
                    &quot;prim&quot;: &quot;int&quot;
                  }
                ]
              },
              {
                &quot;prim&quot;: &quot;code&quot;,
                &quot;args&quot;: [
                  [
                    {
                      &quot;prim&quot;: &quot;CAR&quot;
                    },
                    {
                      &quot;prim&quot;: &quot;PUSH&quot;,
                      &quot;args&quot;: [
                        {
                          &quot;prim&quot;: &quot;int&quot;
                        },
                        {
                          &quot;int&quot;: &quot;1&quot;
                        }
                      ]
                    },
                    {
                      &quot;prim&quot;: &quot;ADD&quot;
                    },
                    {
                      &quot;prim&quot;: &quot;NIL&quot;,
                      &quot;args&quot;: [
                        {
                          &quot;prim&quot;: &quot;operation&quot;
                        }
                      ]
                    },
                    {
                      &quot;prim&quot;: &quot;PAIR&quot;
                    }
                  ]
                ]
              }
            ]
          </div>
        </div>
        <div className=\\"LibraryItem__wrapper2\\" onClick={[Function: onClick]}>
          <div className=\\"LibraryItem__label\\">
            Michelson
          </div>
          <div className=\\"LibraryItem__content\\">
            parameter int;
            storage int;
            code { CAR ;
                   PUSH int 1 ;
                   ADD ;
                   NIL operation ;
                   PAIR }
          </div>
        </div>
      </div>
    </LibraryItem>
  </div>
</Library>"
`;
